src/mapper/pkg/intentsstore/holder.go

```diff
@@ -0,0 +1,149 @@
+package intentsstore
+
+import (
+	"github.com/amit7itz/goset"
+	"github.com/otterize/network-mapper/src/mapper/pkg/cloudclient"
+	"github.com/otterize/network-mapper/src/mapper/pkg/graph/model"
+	"github.com/samber/lo"
+	"sync"
+	"time"
+)
+
+type NamespacedName struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type FullInfoIntentWithTime struct {
+	SourceFullInfo      model.OtterizeServiceIdentity
+	DestinationFullInfo model.OtterizeServiceIdentity
+	Timestamp           time.Time
+	Intent              cloudclient.IntentInput
+}
+
+type SourceDestPair struct {
+	Source      NamespacedName
+	Destination NamespacedName
+}
+
+func serviceIdentityToNameNamespacePair(identity model.OtterizeServiceIdentity) NamespacedName {
+	return NamespacedName{
+		Name:      identity.Name,
+		Namespace: identity.Namespace,
+	}
+}
+
+type DiscoveredIntent struct {
```

Comment:
This will change to just store (discovered) intents instead of source+dest pairs (with potentially some extra info if needed), following yesterday's discussion. I'm documenting it in a separate task.